code/game/objects/items/shooting_range.dm

```diff
@@ -77,101 +76,36 @@
 	desc = "A shooting target that looks like a xenomorphic alien."
 	hp = 2350 // alium onest too kinda
 
-/obj/item/target/bullet_act(var/obj/item/projectile/Proj)
-	var/p_x = Proj.p_x + pick(0,0,0,0,0,-1,1) // really ugly way of coding "sometimes offset Proj.p_x!"
-	var/p_y = Proj.p_y + pick(0,0,0,0,0,-1,1)
-	var/decaltype = 1 // 1 - scorch, 2 - bullet
-
-	if(istype(/obj/item/projectile/bullet, Proj))
-		decaltype = 2
-
-
-	virtualIcon = new(icon, icon_state)
-
-	if( virtualIcon.GetPixel(p_x, p_y) ) // if the located pixel isn't blank (null)
-
-		hp -= Proj.damage
+#define DECALTYPE_SCORCH 1
+#define DECALTYPE_BULLET 2
+
+/obj/item/target/bullet_act(obj/item/projectile/P)
+	var/p_x = P.p_x + pick(0,0,0,0,0,-1,1) // really ugly way of coding "sometimes offset P.p_x!"
+	var/p_y = P.p_y + pick(0,0,0,0,0,-1,1)
+	var/decaltype = DECALTYPE_SCORCH
+	if(istype(P, /obj/item/projectile/bullet))
+		decaltype = DECALTYPE_BULLET
+	var/icon/C = icon(icon, icon_state)
+	if(C.GetPixel(p_x, p_y) && LAZYLEN(overlays) <= 35) // if the located pixel isn't blank (null)
```

Comment:
`if(C.GetPixel(p_x, p_y) && P.original == src && overlays.len <= 35)` is the check TG has. The original check didn't make much sense to me so I kept it out.